crates/re_query/src/util.rs

```diff
@@ -17,14 +17,15 @@ pub fn query_archetype_with_history<'a, A: Archetype + 'a, const N: usize>(
         re_log_types::TimeType::Sequence => history.sequences,
     };
 
-    if visible_history == 0 {
+    if !history.enabled || visible_history == VisibleHistory::OFF {
```

Comment:
How the visible history is stored is not fully normalised: there is a slight duplication:
- the enabled/disabled bool
- the fact that `{from: Relative(0), to: Relative(0)}` (that's what `VisibleHistory::OFF` is) disable the feature for all practical purposes.

I've introduced that "enabled" bool to allow the user to disable the feature without losing their configured boundaries (ie. that checkbox next to the `Visible History` label). Overall, better UX I believe, but the cost is this check.

BTW, the test against `VisibleHistory::OFF` could actually be dropped and the result would be the same. So it can be seen as an optimisation to avoid a range query when we know none is needed :)